python-backend/app/api/users/minespace/resources/minespace_user.py

```diff
@@ -0,0 +1,53 @@
+import uuid
+
+from flask_restplus import Resource, reqparse
+from flask import request
+from app.extensions import api
+from ....utils.access_decorators import requires_role_mine_admin
+from ....utils.resources_mixins import UserMixin, ErrorMixin
+from ..models.minespace_user import MinespaceUser
+from ..models.minespace_user_mine import MinespaceUserMine
+from app.extensions import db
+
+
+class MinespaceUserResource(Resource, UserMixin, ErrorMixin):
+    parser = reqparse.RequestParser()
+    parser.add_argument('email', type=str, location='json', required=True)
+    parser.add_argument('mine_guids', type=list, location='json', required=True)
+
+    @api.doc(params={'user_id': 'User id.'})
+    @requires_role_mine_admin
+    def get(self, user_id=None):
+        if user_id or request.args.get('email'):
+            user = MinespaceUser.find_by_id(user_id)
+            if not user:
+                user = MinespaceUser.find_by_email(request.args.get('email'))
+                if not user:
+                    return self.create_error_payload(500, "user not found"), 500
```

Comment:
Should this be a 500? I think it would be better as a 404.

Reply:
Done